XX-eltoo-transactions.md

```diff
@@ -114,22 +115,24 @@ where
    * `txin[0]` outpoint: `txid` and `output_index` from latest state `k` output (can be 0, the funding output)
    * `txin[0]` sequence: 0xFFFFFFFD, to allow [BIP125](https://github.com/bitcoin/bips/blob/master/bip-0125.mediawiki#summary) replacement
    * `txin[0]` script bytes: 0
-   * `txin[0]` witness: `signature_for_inner_pubkey`
+   * `txin[0]` witness:
+     * control block: 0xc0 marker for tapscript, internel public key, merkle proof unless spending funding tx
+     * tapscript: EXPR_UPDATE_0 or EXPR_UPDATE(locktime), depending on what output is being spent
+     * `signature_for_inner_pubkey`
 * txout count: 1
-   * `txout[0]` amount: the HTLC
+   * `txout[0]` amount: the channel capacity
    * `txout[0]` script: `tr(aggregated_key, {EXPR_UPDATE(locktime+1), EXPR_SETTLE(locktime)})`
-* control block: EXPR_UPDATE_0 or EXPR_UPDATE(locktime) merkle proof, depending on what output is being spent
 
 where EXPR_UPDATE(n) =
 
-`<n> OP_CLTV 1 OP_CHECKSIGVERIFY`
+`1 OP_CHECKSIGVERIFY <n> OP_CLTV`
```

Comment:
`OP_CLTV` and `OP_CSV` leave things on the stack, so either need an `OP_DROP` or should be at the end

Reply:
Why does it matter to leave something on the stack first or last? I just thought it needed to be there at end of script interpreter.

Reply:
`<n> OP_CLTV` leaves `n` at the top of the stack, then `1` is put on top of that, so `CHECKSIGVERIFY` looks at `n` as the sig and `1` as the pubkey.

Reply:
`1` is a wrong here anyways, right? Has to be `OP_1`?

Reply:
maybe consider (also) writing the scripts in miniscript?

Reply:
I'd definitely take a PR that converts everything to miniscript, I'm already aping it a bit in output descriptor style.